aptos-move/aptos-aggregator/src/types.rs

```diff
@@ -59,3 +61,98 @@ impl TryFrom<AggregatorVersionedID> for StateKey {
         }
     }
 }
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum AggregatorValue {
```

Comment:
Having a snapshot inside aggregatorvalue indeed seems confusing. Yeah, changing AggregatorValue to something else might help.